yoti-sdk-auth/src/main/java/com/yoti/auth/AuthenticationTokenGenerator.java

```diff
@@ -0,0 +1,166 @@
+package com.yoti.auth;
+
+import static com.yoti.validation.Validation.notNull;
+import static com.yoti.validation.Validation.notNullOrEmpty;
+
+import java.io.IOException;
+import java.security.KeyPair;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+
+import com.yoti.api.client.InitialisationException;
+import com.yoti.api.client.KeyPairSource;
+import com.yoti.api.client.spi.remote.KeyStreamVisitor;
+import com.yoti.api.client.spi.remote.call.ResourceException;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+
+/**
+ * The {@link AuthenticationTokenGenerator} is used for generation authorization tokens
+ * that can be used for accessing Yoti services.  An authorization token must have
+ * a unique identifier, and an expiry timestamp.  One or more scopes can be provided
+ * to allow the authorization token access to different parts of Yoti systems.
+ * <p>
+ * The {@link AuthenticationTokenGenerator.Builder} can be accessed via {@code AuthorizationTokenGenerator.builder()}
+ * method, and then configured via the fluent API.
+ */
+public class AuthenticationTokenGenerator {
+
+    private static final Map<String, String> DEFAULT_FORM_VALUES;
+
+    static {
+        DEFAULT_FORM_VALUES = new HashMap<>();
+        DEFAULT_FORM_VALUES.put("grant_type", "client_credentials");
+        DEFAULT_FORM_VALUES.put("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
+    }
+
+    private final String sdkId;
+    private final KeyPair keyPair;
+    private final Supplier<String> jwtIdSupplier;
+    private final FormRequestClient formRequestClient;
+    private final SignedJwtGenerator signedJwtGenerator;
+
+    private final String authApiUrl;
+    private final ObjectMapper objectMapper;
```

Comment:
Why is the `ObjectMapper` being injected/exposed?

The `CreateAuthenticationTokenResponse` is defined by this module, shouldn't the `ObjectMapper` (and it's config) _also_ be owned/defined by this module? What would be the point of the client providing their own instance config if they do not control the annotations/layout of `CreateAuthenticationTokenResponse`?